src/CarbonAware.WebApi/src/Filters/HttpResponseExceptionFilter.cs

```diff
@@ -9,62 +9,75 @@ namespace CarbonAware.WebApi.Filters;
             {
-                response = new HttpValidationProblemDetails(){
-                    Title = httpResponseException.Title,
-                    Status = httpResponseException.Status,
-                    Detail = httpResponseException.Detail
-                };
-            } else {
-                var exceptionType = context.Exception.GetType().Name;
-                int statusCode;
-                if (!EXCEPTION_STATUS_CODE_MAP.TryGetValue(exceptionType, out statusCode))
-                {
-                    statusCode = (int)HttpStatusCode.InternalServerError;
-                    activity?.SetStatus(ActivityStatusCode.Error, context.Exception.Message);
-                    _logger.LogError(context.Exception, "500 Error: Unhandled exception");
-                }
-                
-                response = new HttpValidationProblemDetails()
-                {
-                    Title = exceptionType,
-                    Status = statusCode,
-                    Detail = context.Exception.Message
-                };
+                statusCode = (int)HttpStatusCode.InternalServerError;
+                activity?.SetStatus(ActivityStatusCode.Error, context.Exception.Message);
             }
-
-            var traceId = activity?.Id;
-            if (traceId != null)
+            var envVars = _config?.GetSection(CarbonAwareVariablesConfiguration.Key).Get<CarbonAwareVariablesConfiguration>();
+       
+            if (statusCode == (int)HttpStatusCode.InternalServerError &&
+                envVars?.VerboseApi == false)
             {
-                response.Extensions["traceId"] = traceId;
+                 response = new HttpValidationProblemDetails() {
+                                Title = HttpStatusCode.InternalServerError.ToString(),
+                                Status = statusCode,
+                                Detail = context.Exception.Message
+                    };
             }
-
-            context.Result = new ObjectResult(response)
+            else
             {
-                StatusCode = response.Status
-            };
+                response = new HttpValidationProblemDetails() {
+                            Title = exceptionType,
+                            Status = statusCode,
+                            Detail = context.Exception.Message
+                };
+                if (envVars?.VerboseApi == true) {
+                    response.Errors["stackTrace"] = new string[] { context.Exception.StackTrace! };
+                }
+            }
+        }
 
-            context.ExceptionHandled = true;
+        var traceId = activity?.Id;
+        if (traceId != null)
+        {
+            response.Extensions["traceId"] = traceId;
         }
+
+        context.Result = new ObjectResult(response)
+        {
+            StatusCode = response.Status
+        };
+        _logger.LogError(context.Exception, "Error OnException");
```

Comment:
Suggestion:  Use structured logging to also output the message.  For example:  _logger.LogError(context.Exception, "Exception:  {exception}", context.Exception.Message);